cosmosdb/cosmosdb_utils.bal

```diff
@@ -0,0 +1,128 @@
+//
+// Copyright (c) 2018, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 Inc. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+//
+
+import ballerina/time;
+import ballerina/http;
+import ballerina/crypto;
+import ballerina/system;
+
+string dateString = "";
+
+function costructRequestHeaders(http:Request request, string httpMethod, string resourceType, string resourceLink,
+                                string masterKey, string contentType, RequestOptions? requestOptions = ()) returns error
+                                                                                                                   ? {
+    time:Time time = time:currentTime();
+    time:Timezone zoneValue = { zoneId: "GMT" };
+    time:Time standardTime = new(time.time, zoneValue);
+    dateString = standardTime.format(DATE_TIME_FORMAT);
+
+    string stringToSign = httpMethod.toLower() + NEW_LINE + resourceType.toLower() + NEW_LINE + resourceLink
+        + NEW_LINE + dateString.toLower() + NEW_LINE + EMPTY_STRING + NEW_LINE;
+    string signature = crypto:hmac(stringToSign, masterKey, keyEncoding = crypto:BASE64,
+        crypto:SHA256).base16ToBase64Encode();
+    string authHeaderString = check http:encode("type=" + MASTER_TOKEN + "&ver=" + TOKEN_VERSION + "&sig=" + signature,
+        "UTF-8");
+    request.setHeader("Authorization", authHeaderString);
+    request.setHeader("x-ms-version", X_MS_VERSION);
+    request.setHeader("x-ms-date", dateString);
+    request.setHeader("content-type", contentType);
```

Comment:
Some headers are defined as constants in cosmodb_constants.bal. Shall we do the same for all these headers as well?